etc/tests/test_bootstrapper.py

```diff
@@ -172,10 +172,54 @@ def is_writable_dir(path):
         return False
 
 
+def remove_file(filename, fail_ok=True):
+    """Removes filename. If fail_ok is False an assert is raised
+       if removal failed for any reason, e.g. filenotfound
+    """
+    try:
+        os.remove(filename)
+    except OSError as ose:
+        if fail_ok is False:
+            raise AssertionError('Cannot remove {}: {} {}'
+                                 .format(filename,
+                                         str(ose),
+                                         ose))
+
+
+def test_process_metrics_method_not_writable_dir(monkeypatch, s3_setup, tmpdir):
+    """Test for process_metrics_and_metadata
+
+    Validates that the method can handle output directory that is not writable
+    """
+
+    # remove "default" output file if it already exists
+    output_metadata_file = Path('/tmp') / 'mlpipeline-ui-metadata.json'
+    remove_file(output_metadata_file)
+
+    try:
+        os.environ['ELYRA_WRITABLE_CONTAINER_DIR'] = '/does/not/exist'
```

Comment:
I'm curious why `monkeypatch` wasn't used here?  Was this intentional?  If so, could you add a comment as to why?

Reply:
A bug